File: examples/js/src/002-translation-ollama.mjs
import fs from "fs/promises";
import { ChatOllama } from "@langchain/ollama";

const model = new ChatOllama({
  model: "translategemma:12b",
  think: false
});

async function translate(text, language) {
  const response = await model.invoke([
    {
      role: "system",
      content: `You are a helpful assistant that translates text from English to German.
      Translate the given text to German. Return only the translated text, no other text.`,
    },
    {
      role: "user",
      content: text,
    },
  ]);
  return response.content;
}

async function updateTranslations() {
  const translations = {
    en: JSON.parse(await fs.readFile("public/locales/en.json", "utf8")),
    de: JSON.parse(await fs.readFile("public/locales/de.json", "utf8")),
  }

  console.log(translations);

  await Promise.all(Object.keys(translations.en).map(async (key) => {
    const enTranslation = translations.en[key];
    if (!translations.de[key]) {
        translations.de[key] = await translate(enTranslation, "de");
      }
    }));

  console.log(translations);
  // TODO: implement write to file
}

updateTranslations();

File: examples/python/src/002_translation_ollama.py
import json
import asyncio
from langchain_ollama import ChatOllama

model = ChatOllama(model="qwen3.5:latest", think=False)

async def translate(text: str, language: str) -> str:
    response = await model.ainvoke([
        {
            "role": "system",
            "content": "You are a helpful assistant that translates text from English to German.\n"
                       "Translate the given text to German. Return only the translated text, no other text.",
        },
        {
            "role": "user",
            "content": text,
        },
    ])
    return response.content


async def update_translations():
    with open("locales/en.json", "r") as f:
        en = json.load(f)
    with open("locales/de.json", "r") as f:
        de = json.load(f)

    tasks = []
    keys_to_translate = []
    for key in en:
        if key not in de:
            keys_to_translate.append(key)
            tasks.append(translate(en[key], "de"))

    results = await asyncio.gather(*tasks)
    for key, result in zip(keys_to_translate, results):
        de[key] = result

    print(json.dumps({"en": en, "de": de}, indent=4))
    # TODO: implement write to file


asyncio.run(update_translations())
